pkg/controller/servicebindingrequest/sbrcontroller.go

```diff
@@ -84,29 +92,71 @@ func (s *SBRController) getWatchingGVKs() ([]schema.GroupVersionKind, error) {
 	return append(gvks, olmGVKs...), nil
 }
 
+func createUpdateFuncWithLog(l *log.Log) func(updateEvent event.UpdateEvent) bool {
+	return func(e event.UpdateEvent) bool {
+		isSecret := isOfKind(e.ObjectNew, "Secret")
+		isConfigMap := isOfKind(e.ObjectNew, "ConfigMap")
+
+		if isSecret || isConfigMap {
+			var (
+				dataFieldsAreEqual bool
+				err                error
+			)
+
+			if dataFieldsAreEqual, err = compareObjectFields(e.ObjectNew, e.ObjectOld, "data"); err != nil {
+				l.Error(err, "")
+				return false
+			}
+
+			shouldReconcile := !dataFieldsAreEqual
+
+			l.Debug("Predicate evaluated", "ShouldReconcile", shouldReconcile)
+			return shouldReconcile
+		}
+
+		// ignore updates to CR status in which case metadata.Generation does not change
+		return e.MetaOld.GetGeneration() != e.MetaNew.GetGeneration()
+	}
+}
+
+func newGVKPredicate(l *log.Log) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: createUpdateFuncWithLog(l),
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			// evaluates to false if the object has been confirmed deleted
+			return !e.DeleteStateUnknown
+		},
+	}
+}
+
 // AddWatchForGVK creates a watch on a given GVK, as long as it's not duplicated.
 func (s *SBRController) AddWatchForGVK(gvk schema.GroupVersionKind) error {
-	log := s.logger.WithValues("GVK", gvk)
-	log.Debug("Adding watch for GVK...")
+	l := s.logger.WithValues("GVK", gvk)
```

Comment:
@isutton also here, name it `logger` instead of `l`.

Reply:
Solved in b3b8a636334c619945d6d816d1d3122c02312d2f.